src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabHookCreator.java

```diff
@@ -145,17 +145,31 @@ public static void createSystemHookWhenMissing(GitLabServer server,
         }
     }
 
+    /**
+     * @deprecated use {@link getHookUrl(String,boolean)} instead
+     */
+    @Deprecated
     public static String getHookUrl(boolean isWebHook) {
-        String rootUrl = Jenkins.get().getRootUrl();
+        return getHookUrl(null, isWebHook);
+    }
+
+    /**
+     * @param hooksRootUrl the root URL to use in the hook URL.
+     *        If {@code null} or empty, {@link Jenkins#getRootUrl()} will be used instead.
+     * @param isWebHook {@code true} to get the webhook URL, {@code false} for the systemhook URL
+     * @return a webhook or systemhook URL
+     */
+    public static String getHookUrl(String hooksRootUrl, boolean isWebHook) {
+        String rootUrl = StringUtils.isBlank(hooksRootUrl) ? Jenkins.get().getRootUrl() : hooksRootUrl;
         if (StringUtils.isBlank(rootUrl)) {
             return "";
         }
         checkURL(rootUrl);
-        UriTemplateBuilder templateBuilder = UriTemplate.buildFromTemplate(rootUrl);
+        UriTemplateBuilder templateBuilder = UriTemplate.buildFromTemplate(StringUtils.stripEnd(rootUrl, "/"));
```

Comment:
The `Jenkins.get().getRootUrl()` ensures stripped root url.

I believe we are best off stripping it on the getter and no here.
```suggestion
        UriTemplateBuilder templateBuilder = UriTemplate.buildFromTemplate(rootUrl);
```

Reply:
`Jenkins.get().getRootUrl()` actually ensure that there is a trailing slash, so I did something similar in 104fd92156cc4f2e4da0f4d35b7cf5ad1fb3c4d2: `getHooksRootUrl()` now returns either an URL with trailing slash, or `null`.

If you merge this PR, could you squash these two commits?

Reply:
Squash commit is always the preferred option in this repo, see past commits 👍

Reply:
Ensuring it ends on `/` is also fine 👏